test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -287,26 +290,42 @@ TEST_F(XdsOverrideHostTest, DrainingSubchannelIsConnecting) {
   ApplyUpdateWithHealthStatuses(
       {{kAddresses[0], XdsHealthStatus::HealthStatus::kUnknown},
        {kAddresses[1], XdsHealthStatus::HealthStatus::kDraining},
-       {kAddresses[2], XdsHealthStatus::HealthStatus::kHealthy}});
+       {kAddresses[2], XdsHealthStatus::HealthStatus::kHealthy}},
+      {"UNKNOWN", "HEALTHY", "DRAINING"});
   auto subchannel = FindSubchannel(kAddresses[1]);
   ASSERT_NE(subchannel, nullptr);
-  subchannel->SetConnectivityState(GRPC_CHANNEL_IDLE);
+  // There are two notifications - one from child policy and one from the parent
+  // policy due to draining channel update
   picker = ExpectState(GRPC_CHANNEL_READY);
-  ExpectPickQueued(picker.get(), pick_arg);
+  EXPECT_EQ(ExpectPickComplete(picker.get(), pick_arg), kAddresses[1]);
+  ExpectRoundRobinPicks(picker.get(), {kAddresses[0], kAddresses[2]});
+  subchannel->SetConnectivityState(GRPC_CHANNEL_IDLE);
+  ExpectStateAndQueuingPicker(GRPC_CHANNEL_READY, absl::OkStatus(), pick_arg);
```

Comment:
I think we need to check the behavior both with the `pick_arg` and without.  Suggest replacing this line with:

```
picker = ExpectState(GRPC_CHANNEL_READY);
ExpectPickQueued(picker.get(), pick_arg);
ExpectRoundRobinPicks(picker.get(), {kAddresses[0], kAddresses[2]});
```